src/components/Footer.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Todo } from '../types/Todo';
+import { Selected } from '../types/Selected';
+import { ItemsLeft } from './ItemsLeft';
+
+interface Props {
+  allTodos: Todo[];
+  selected: Selected;
+  setSelected: (selected: Selected) => void;
+  clearAll: () => void;
+  completedTodos: number;
+}
+
+export const Footer: React.FC<Props> = ({
+  allTodos,
+  selected,
+  setSelected,
+  clearAll,
+  completedTodos,
+}) => {
+  const filters = [
+    { title: 'All', type: Selected.All, href: '#/' },
+    { title: 'Active', type: Selected.Active, href: '#/active' },
+    { title: 'Completed', type: Selected.Completed, href: '#/completed' },
+  ];
```

Comment:
it can be constant outside of component. in component such array will be created each render.